systest/backup/common/utils.go

```diff
@@ -91,6 +105,257 @@ func CopyToLocalFs(t *testing.T) {
+		Query: checkData,
+		Variables: map[string]interface{}{
+			"name": "Item" + strconv.Itoa(desriedSuffix),
+		},
+	}
+
+	b, err := json.Marshal(params)
+	adminUrl := "http://" + testutil.ContainerAddr(whichAlpha, 8080) + "/graphql"
+	require.NoError(t, err)
+
+	req, err := http.NewRequest(http.MethodPost, adminUrl, bytes.NewBuffer(b))
+	req.Header.Set("Content-Type", "application/json")
+	if jwtToken != "" {
+		req.Header.Set(accessJwtHeader, jwtToken)
+	}
+	client := &http.Client{}
+	resp, err := client.Do(req)
+
+	var data interface{}
+	require.NoError(t, json.NewDecoder(resp.Body).Decode(&data))
+}
+
+func TakeBackup(t *testing.T, jwtToken string, backupDst string, whichAlpha string) {
+
+	backupRequest := `mutation backup($dst: String!) {
+		backup(input: {destination: $dst}) {
+			response {
+				code
+			}
+			taskId
+		}
+	}`
+	params := testutil.GraphQLParams{
+		Query: backupRequest,
+		Variables: map[string]interface{}{
+			"dst": backupDst,
+		},
+	}
+
+	b, err := json.Marshal(params)
+	adminUrl := "http://" + testutil.ContainerAddr(whichAlpha, 8080) + "/admin"
+	require.NoError(t, err)
+
+	req, err := http.NewRequest(http.MethodPost, adminUrl, bytes.NewBuffer(b))
+	req.Header.Set("Content-Type", "application/json")
+	if jwtToken != "" {
+		req.Header.Set(accessJwtHeader, jwtToken)
+	}
+	client := &http.Client{}
+	resp, err := client.Do(req)
+
+	var data interface{}
+	require.NoError(t, json.NewDecoder(resp.Body).Decode(&data))
+
+	require.Equal(t, "Success", testutil.JsonGet(data, "data", "backup", "response", "code").(string))
+	taskId := testutil.JsonGet(data, "data", "backup", "taskId").(string)
+	testutil.WaitForTask(t, taskId, false, testutil.ContainerAddr(whichAlpha, 8080))
+
+	defer changeFolderPermission(t, whichAlpha)
+}
+
+func changeFolderPermission(t *testing.T, whichAlpha string) {
+
+	cmd := []string{"bash", "-c", "chmod -R 777 /data/backups"}
+	require.NoError(t, testutil.DockerExec(whichAlpha, cmd...))
+
+}
+
+func RunRestore(t *testing.T, jwtToken string, restoreLocation string, whichAlpha string) {
+	restoreRequest := `mutation restore($loc: String!) {
+		restore(input: {location: $loc}) {
+				code
+				message
+			}
+	}`
+	params := testutil.GraphQLParams{
+		Query: restoreRequest,
+		Variables: map[string]interface{}{
+			"loc": restoreLocation,
+		},
+	}
+
+	b, err := json.Marshal(params)
+	adminUrl := "http://" + testutil.ContainerAddr(whichAlpha, 8080) + "/admin"
+	require.NoError(t, err)
+
+	req, err := http.NewRequest(http.MethodPost, adminUrl, bytes.NewBuffer(b))
+	req.Header.Set("Content-Type", "application/json")
+	if jwtToken != "" {
+		req.Header.Set(accessJwtHeader, jwtToken)
+	}
+	client := &http.Client{}
+	resp, err := client.Do(req)
+
+	var data interface{}
+	require.NoError(t, json.NewDecoder(resp.Body).Decode(&data))
+
+	require.Equal(t, "Success", testutil.JsonGet(data, "data", "restore", "code").(string))
+}
+
+func WaitForRestore(t *testing.T, whichAlpha string) {
```

Comment:
what is the difference in this function and `testutil.WaitForRestore` function?

Reply:
Resolved.

Due to the small amount of restored data, I am skipping the drain mode check in the new function.

I am now using already existing `testutil.WaitForRestore`.